### PR TITLE
chore: add workflows

### PR DESCRIPTION
**Pull request by Phil91**
## Description

add workflows to check the quality of the code

## Why

To add quality checks to the repo

## Issue

#2 

## Checklist

- [x] I have followed the contributing guidelines
- [x] I have performed [IP checks](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04#checking-libraries-using-the-eclipse-dash-license-tool) for added or updated 3rd party libraries
- [x] I have created and linked IP issues or requested their creation by a committer
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: .github/workflows/sonarcloud.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        run: |
+          mkdir -p ./.sonar/scanner
+          dotnet tool update dotnet-sonarscanner --tool-path ./.sonar/scanner
+      - name: Build and analyze
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: |
+          dotnet tool install --global dotnet-coverage
+          ./.sonar/scanner/dotnet-sonarscanner begin /k:"${{ vars.SONAR_PROJECT_KEY }}" /o:"${{ vars.SONAR_ORGANIZATION }}" /d:sonar.login="${{ secrets.SONAR_TOKEN }}" /d:sonar.host.url="https://sonarcloud.io" /d:sonar.cs.vscoveragexml.reportsPaths=src/coverage.xml
```
Comment (evegufy) [inline .github/workflows/sonarcloud.yml:70]: did you request the setup of the vars/secret?

File: docs/technical-documentation/dev-process/How to contribute.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
+# Contribution details
+
+To contribute to the policy hub as part of the open source community, please read the details defined below.
```
Comment (evegufy) [inline docs/technical-documentation/dev-process/How to contribute.md:3]: Please remove policy hub

File: .github/workflows/chart-release.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        uses: helm/chart-releaser-action@v1.4.1
+        env:
+          CR_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+          CR_SKIP_EXISTING: "true"
+
+      - name: Push git tag for release workflow to be triggered
+        uses: rickstaa/action-create-tag@a1c7777fcb2fee4f19b0f283ba888afa11678b72 # v1.7.2
+        with:
+          tag: v${{ steps.chart-release.outputs.chart_version }}
+        if: ${{ steps.chart-release.outputs.changed_charts }}
```
Comment (evegufy) [inline .github/workflows/chart-release.yaml:70]: Please change to [changed version using git push](https://github.com/eclipse-tractusx/policy-hub/pull/63/files)

File: docs/technical-documentation/dev-process/Dev-flow_deploy-dev-env.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+```mermaid
+flowchart LR
+    subgraph local
+    D(Developer)
+    end
+    subgraph eclipse-tractusx
+        direction LR
+        D -- PR* to dev*--> PH(policy-hub**)
+        click PH "https://github.com/eclipse-tractusx/policy-hub"
```
Comment (evegufy) [inline docs/technical-documentation/dev-process/Dev-flow_deploy-dev-env.md:11]: please change to this repo

File: docs/technical-documentation/dev-process/Dev-flow_deploy-dev-env.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    end
+    subgraph Argo CD - sync to k8s cluster
+    PH -- auto-sync --> A(Argo CD dev)
+    click A "https://argo.dev.demo.catena-x.net"
```
Comment (evegufy) [inline docs/technical-documentation/dev-process/Dev-flow_deploy-dev-env.md:15]: Please remove this link, as link to argo dev not sustainable and referencing consortia specific env

File: docs/technical-documentation/dev-process/Dev-flow_git-diagram.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+    merge bug/bug3
+    merge dev
+    checkout dev
+    branch feature/feature5 order: 11
+    commit id: "feat(function): add feature5"
+    checkout bug/bug2
+    commit id:"fix(function): change2 bug2"
+    checkout release/1.2.0
+    merge bug/bug2
+    commit id: "release(1.2.0-RC2): update version, changelog..." tag: "1.2.0-RC2"
```
Comment (evegufy) [inline docs/technical-documentation/dev-process/Dev-flow_git-diagram.md:74]: please change to 1.2.0-rc.2

File: docs/technical-documentation/dev-process/Dev-flow_git-diagram.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    commit id: "feat(function)!: change feature2"
+    checkout dev
+    merge feature/feature2
+    checkout feature/feature4
+    commit id: "feat(function): change feature4"
+    checkout dev
+    merge feature/feature4
+    branch release/1.2.0 order: 8
+    commit id: "release(1.2.0-RC1): prepare migration (backend)"
+    commit id: "release(1.2.0-RC1): update version, changelog..." tag: "1.2.0-RC1"
```
Comment (evegufy) [inline docs/technical-documentation/dev-process/Dev-flow_git-diagram.md:53]: please change to 1.2.0-rc.1

File: docs/technical-documentation/dev-process/Enumeration Handling.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
+## Enumeration
+
```
Comment (evegufy) [inline docs/technical-documentation/dev-process/Enumeration Handling.md:2]: this entire file seems quite policy hub specific

**Review by evegufy — CHANGES_REQUESTED** [review]

**Comment by Phil91** [comment]
sonar setup was requested via eclipse gitlab:

https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/4403

File: scripts/license.sh [hunk 9]
```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+###############################################################
+# Copyright (c) 2021, 2023 Contributors to the Eclipse Foundation
```
[the remark this answers is not in the document]
Comment (Phil91) [reply scripts/license.sh]: done

File: scripts/check-dependencies.md [hunk 10]
```diff
@@ -0,0 +1,9 @@
+# Check dependencies
+
+Dependencies are checked by the [Eclipse Dash License Tool](https://github.com/eclipse/dash-licenses) with a GitHub workflow (dependencies.yaml).
+
+This workflow uses the executable jar in the download directory.
+
+In order to update the executable jar run the following command from the root directory:
+
+    curl -L --output ./scripts/download/org.eclipse.dash.licenses-1.1.1.jar 'https://repo.eclipse.org/service/local/artifact/maven/redirect?r=dash-licenses&g=org.eclipse.dash&a=org.eclipse.dash.licenses&v=1.1.1'
```
[the remark this answers is not in the document]
Comment (Phil91) [reply scripts/check-dependencies.md:9]: done

File: scripts/add_notice_footer.sh [hunk 11]
```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+###############################################################
+# Copyright (c) 2021, 2023 Contributors to the Eclipse Foundation
```
[the remark this answers is not in the document]
Comment (Phil91) [reply scripts/add_notice_footer.sh]: done

File: .github/workflows/chart-test.yml [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+      node_image:
+        description: 'kindest/node image for k8s kind cluster'
+        # k8s version from 3.1 release as default
+        default: 'kindest/node:v1.27.3'
+        required: false
+        type: string
+      upgrade_from:
+        description: 'ssi credential issuer chart version to upgrade from'
+        # tbd
+        default: '0.1.0-rc.2'
```
[the remark this answers is not in the document]
Comment (Phil91) [reply .github/workflows/chart-test.yml]: done

File: .github/workflows/chart-test.yml [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+        if: github.event_name != 'pull_request' || steps.list-changed.outputs.changed == 'true'
+
+      # TODO: re-add the step after the first version release  
+      # Upgrade the released chart version with the locally available chart
+      # default value for event_name != workflow_dispatch
+      # - name: Run helm upgrade
+      #   run: |
+      #     helm repo add bitnami https://charts.bitnami.com/bitnami
+      #     helm repo add tractusx-dev https://eclipse-tractusx.github.io/charts/dev
+      #     helm install issuer tractusx-dev/issuer --version ${{ github.event.inputs.upgrade_from || '0.1.0-rc.2' }} --namespace upgrade --create-namespace
```
[the remark this answers is not in the document]
Comment (Phil91) [reply .github/workflows/chart-test.yml]: done

File: .github/workflows/chart-release.yaml [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+      - name: Install Helm
+        uses: azure/setup-helm@v3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Update helm dependencies for issuer
```
[the remark this answers is not in the document]
Comment (Phil91) [reply .github/workflows/chart-release.yaml]: done

File: .github/workflows/chart-release.yaml [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+        run: |
+          cd charts/issuer
```
[the remark this answers is not in the document]
Comment (Phil91) [reply .github/workflows/chart-release.yaml]: done

File: .github/workflows/chart-test.yml [hunk 16]
```diff
@@ -0,0 +1,119 @@
...
+#
+# SPDX-License-Identifier: Apache-2.0
+###############################################################
+
+name: Lint and Test Chart
+
+on:
+  push:
+    paths:
+      - 'charts/issuer/**'
```
[the remark this answers is not in the document]
Comment (Phil91) [reply .github/workflows/chart-test.yml]: done

File: .github/workflows/chart-test.yml (hunk 13 above)
[the remark this answers is not in the document]
Comment (Phil91) [reply .github/workflows/chart-test.yml]: done

File: .github/workflows/chart-test.yml [hunk 17]
```diff
@@ -0,0 +1,119 @@
...
+          changed=$(ct list-changed --target-branch ${{ github.event.repository.default_branch }})
+          if [[ -n "$changed" ]]; then
+            echo "changed=true" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Run chart-testing (lint)
+        run: ct lint --validate-maintainers=false --check-version-increment=false --target-branch ${{ github.event.repository.default_branch }}
+
+      - name: Run service chart-testing (install)
+        run: ct install --charts charts/issuer --config charts/chart-testing-config.yaml --helm-extra-set-args "--set=issuer.image=kind-registry:5000/service:testing --set=issuermigrations.image=kind-registry:5000/migrations:testing"
```
[the remark this answers is not in the document]
Comment (Phil91) [reply .github/workflows/chart-test.yml]: done

File: .github/workflows/owasp-zap.yml [hunk 18]
```diff
@@ -0,0 +1,138 @@
...
+          helm repo add bitnami https://charts.bitnami.com/bitnami 
+          helm repo update
+
+      - name: Update Helm dependencies
+        run: |
+          cd charts/issuer
+          helm dependency build
+
+      - name: Install the chart on KinD cluster
+        run: helm install testing -n apps --create-namespace --wait --set issuer.image=kind-registry:5000/service:testing --set=issuermigrations.image=kind-registry:5000/migrations:testing --set=issuer.swaggerEnabled=true charts/issuer
```
[the remark this answers is not in the document]
Comment (Phil91) [reply .github/workflows/owasp-zap.yml]: done

File: docs/technical-documentation/dev-process/How to contribute.md (hunk 2 above)
Comment (Phil91) [reply docs/technical-documentation/dev-process/How to contribute.md]: done

File: .github/workflows/chart-release.yaml (hunk 3 above)
Comment (Phil91) [reply .github/workflows/chart-release.yaml]: done

File: docs/technical-documentation/dev-process/Dev-flow_deploy-dev-env.md (hunk 4 above)
Comment (Phil91) [reply docs/technical-documentation/dev-process/Dev-flow_deploy-dev-env.md]: done

File: docs/technical-documentation/dev-process/Dev-flow_deploy-dev-env.md (hunk 5 above)
Comment (Phil91) [reply docs/technical-documentation/dev-process/Dev-flow_deploy-dev-env.md]: done

File: docs/technical-documentation/dev-process/Dev-flow_git-diagram.md (hunk 6 above)
Comment (Phil91) [reply docs/technical-documentation/dev-process/Dev-flow_git-diagram.md]: done

File: docs/technical-documentation/dev-process/Dev-flow_git-diagram.md (hunk 7 above)
Comment (Phil91) [reply docs/technical-documentation/dev-process/Dev-flow_git-diagram.md]: done

File: docs/technical-documentation/dev-process/Enumeration Handling.md (hunk 8 above)
Comment (Phil91) [reply docs/technical-documentation/dev-process/Enumeration Handling.md]: removed

File: .github/workflows/sonarcloud.yml [hunk 19]
```diff
@@ -0,0 +1,75 @@
...
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+###############################################################
+
+name: Sonarcloud
+on:
+  push:
```
Comment (evegufy) [inline .github/workflows/sonarcloud.yml:22]: maybe you could still limit the triggering to src path like [here](https://github.com/eclipse-tractusx/ssi-credential-issuer/pull/8/files#diff-d7e1c74eda3e35f5cbf4cdfbefab1a0602329871a6f5e76f42d4335815a9a289R23)

**Review by evegufy — CHANGES_REQUESTED** [review]

File: .github/workflows/sonarcloud.yml (hunk 19 above)
Comment (Phil91) [reply .github/workflows/sonarcloud.yml:22]: added the path filter to the workflow

**Review by evegufy — APPROVED** [review]

**Pull request merged**